pandas/tests/io/pytables/test_read.py

```diff
@@ -42,6 +42,20 @@ def test_read_missing_key_close_store(tmp_path, setup_path):
     df.to_hdf(path, "k2")
 
 
+def test_read_index_error_close_store(tmp_path, setup_path):
+    # GH 25766
+    path = tmp_path / setup_path
+    df = pd.DataFrame({"A": [], "B": []}, index=[])
+    df.to_hdf(path, "k1")
+
+    with pytest.raises(IndexError, match=r"list index out of range"):
+        read_hdf(path, "k1", stop=0)
+
+    # smoke test to test that file is properly closed after
+    # read with IndexError before another write
+    df.to_hdf(path, "k1")
```

Comment:
```suggestion
    # smoke test to test that file is properly closed after
    # read with IndexError before another write
    with tm.assert_produces_warning(None):
        df.to_hdf(path, "k1")
```

Reply:
why would this produce a warning? If the file is already open an exception is raised regardless of warning filter

Reply:
this change would need to be in test_read_missing_key_opened_store also

Reply:
This is just to help assert something additional with this call (that no warning is raised)

Reply:
I don't think it's correct to do that here